pkg/codegen/convert/plugin_mapper.go

```diff
@@ -93,6 +93,11 @@ func ProviderFactoryFromHost(host plugin.Host) ProviderFactory {
 type mapperPluginSpec struct {
 	name    tokens.Package
 	version semver.Version
+	// An optional list of providers this plugin can map to, only filled if GetMappings is implemented.
+	mappings []string
+	// Set to true once we've called GetMappings, mappings may still be nil after this if GetMappings wasn't
+	// implemented.ß
+	calledGetMappings bool
```

Comment:
I have a bit of mixed feelings about `calledGetMappings`, from the code I can see that it is a workaround for not being able to tell the difference in empty `mappings` vs. no `mappings` (another example of missing the option type in Go) 

Can we maybe change it to something like `hasMappings` instead? I think we care more about the existence of mappings rather than whether we called the function or not

Reply:
Yeh that sounds good

Reply:
I re-looked at this and I'm thinking "called" is a better name because we don't want to recall GetMappings, but "hasMappings" would look false if GetMappings had returned nil.